python/cudf/cudf/pylibcudf_tests/test_lists.py

```diff
@@ -134,3 +141,35 @@ def test_index_of_list_column(test_data, column):
     expect = pa.array(column[1], type=pa.int32())
 
     assert_column_eq(expect, res)
+
+
+@pytest.mark.parametrize(
+    "set_operation,expected",
+    [
+        (
+            plc.lists.difference_distinct,
+            [[], [1, 2, 3], None, [4, 5]],
+        ),
+        (
+            plc.lists.have_overlap,
+            [True, False, None, True],
+        ),
+        (
+            plc.lists.intersect_distinct,
+            [[1, 2], [], None, [None]],
+        ),
+        (
+            plc.lists.union_distinct,
+            [[2, 1, 3], [1, 2, 3, 4, 5], None, [4, None, 5]],
+        ),
+    ],
+)
+def test_set_operations(set_lists_column, set_operation, expected):
+    lhs, rhs = set_lists_column
+
+    res = set_operation(
+        plc.interop.from_arrow(pa.array(lhs)),
+        plc.interop.from_arrow(pa.array(rhs)),
```

Comment:
Can you parametrize this test for nulls_equal/nans_equal as well?

Reply:
Yeah, I can for `nulls_equal`. I think`nans_equal` is a bit trickier to test because of how [cudf handles missing data](https://docs.rapids.ai/api/cudf/stable/user_guide/missing-data/). Can you think of any test columns I can use (lhs and rhs) to test `nan_equal=False`? 

For now I left all `nan_equal` argument as True.

Reply:
Oh do you mean that cudf will convert nan values like np.nan to <NA>?

Reply:
Not sure what the issue was before, but it works now.